frontend/components/Domain/Recipe/RecipeDialogTimeline.vue

```diff
@@ -0,0 +1,279 @@
+                        @delete="deleteTimelineEvent(index)"
+                      />
+                    </v-col>
+                  </v-row>
+                </v-card-title>
+                <v-card-text v-if="event.eventMessage">
+                  {{ event.eventMessage }}
+                </v-card-text>
+              </v-sheet>
+            </v-card>
+          </v-timeline-item>
+        </v-timeline>
+      </v-card>
+      <v-card v-else>
+        <v-card-title class="justify-center">
+          {{ $t("recipe.timeline-is-empty") }}
+        </v-card-title>
+      </v-card>
+    </div>
+
+    <!-- Mobile | Dense -->
+    <div v-else>
+      <v-card
+        v-if="timelineEvents && timelineEvents.length"
+        height="fit-content"
+        max-height="70vh"
+        width="100%"
+        style="overflow-y: auto;"
+      >
+        <v-timeline dense>
+          <v-timeline-item
+            v-for="(event, index) in timelineEvents"
+            :key="event.id"
+            class="pr-3 mb-9"
+            fill-dot
+            small
+            :icon="chooseEventIcon(event)"
+          >
+            <v-row v-if="event.timestamp" no-gutters class="mb-1">
+              <v-col align-self="center" cols="auto">
+                <UserAvatar :user-id="event.userId" />
+              </v-col>
+              <v-col align-self="center" class="ml-3">
+                <v-chip label small>
+                  <v-icon> {{ $globals.icons.calendar }} </v-icon>
+                  {{ new Date(event.timestamp+"Z").toLocaleDateString($i18n.locale) }}
+                </v-chip>
+              </v-col>
+              <v-spacer />
+              <v-col cols="auto">
+                <RecipeTimelineContextMenu
+                  v-if="$auth.user && $auth.user.id == event.userId && event.eventType != 'system'"
+                  :menu-top="false"
+                  :slug="slug"
+                  :event="event"
+                  :menu-icon="$globals.icons.dotsVertical"
+                  fab
+                  color="transparent"
+                  :elevation="0"
+                  :card-menu="false"
+                  :use-items="{
+                    edit: true,
+                    delete: true,
+                  }"
+                  @update="updateTimelineEvent(index)"
+                  @delete="deleteTimelineEvent(index)"
+                />
+              </v-col>
+            </v-row>
+            <v-row no-gutters class="mt-0">
+              <v-col>
+                <strong>{{ event.subject }}</strong>
+                <div v-if="event.eventMessage" class="text-caption">
+                  {{ event.eventMessage }}
+                </div>
+              </v-col>
+            </v-row>
+          </v-timeline-item>
+        </v-timeline>
+      </v-card>
+      <v-card v-else>
+        <v-card-text class="justify-center pa-1">
+          {{ $t("recipe.timeline-is-empty") }}
+        </v-card-text>
+      </v-card>
+    </div>
```

Comment:
A lot of this seems like duplicated code with a few attributes changed if mobile or not. Could we consolidate these a bit and use a computed property that returns and object that has the attributes we need for mobile/desktop?

Something like this?

```typescript

const attrs = computed(() => {
    if (mobile) {
       return { card: { dense: true } /* other stuff */ }
    }

    return { card: { dense: false } /* other stuff */ }
})

```

Reply:
The differences between the two are more coupled than just attributes, but really that just means I need to clean it up more haha. Will work on this!